node_modules/oae-authentication/lib/api.js

```diff
@@ -307,12 +308,31 @@ var _getOrCreateUser = function(ctx, loginId, displayName, opts, callback) {
                     opts.emailVerified = true;
                 }
 
-                createUser(ctx, loginId, displayName, opts, function(err, user) {
+                // If an email address was provided by a non authoritative source such as local user
+                // sign up or a Google / Twitter sign in we should check whether the email address
+                // matches the tenant's configured email domain
+                var shouldCheckEmail = (opts.email && !opts.authoritative);
+
+                // However, if a user followed a link from an invitation email we do not check whether
+                // the email belongs to the configured tenant's email domain. This is to allow for the
+                // situation where a tenant's email domain gets changed after a user gets invited
+                // but before the user has accepted the invitation
+                if (opts.invitationToken && opts.emailVerified) {
+                    shouldCheckEmail = false;
+                }
+
+                OaeUtil.invokeIfNecessary(shouldCheckEmail, _validateEmailBelongsToTenant, ctx, opts.email, function(err) {
```

Comment:
I don't understand how that is expanding the scope of this PR. If someone logs in with GMail, doesn't this PR block their signin too with the `validateEmailBelongsToTenant` function?